src/Compilers/CSharp/Portable/Symbols/NativeIntegerTypeSymbol.cs

```diff
@@ -93,50 +172,239 @@ internal NativeIntegerTypeSymbol(NamedTypeSymbol underlyingType) : base(underlyi
+        {
+            Debug.Assert(!underlyingMethod.IsGenericMethod);
+            _container = container;
+            _associatedSymbol = associatedSymbol;
+            UnderlyingMethod = underlyingMethod;
+            NativeIntegerTypeSymbol.VerifyEquality(this, underlyingMethod);
+        }
+
+        public override Symbol ContainingSymbol => _container;
+
+        public override MethodSymbol UnderlyingMethod { get; }
+
+        public override TypeWithAnnotations ReturnTypeWithAnnotations => _container.SubstituteUnderlyingType(UnderlyingMethod.ReturnTypeWithAnnotations);
+
+        public override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotations => ImmutableArray<TypeWithAnnotations>.Empty;
 
-                return ImmutableArray<NamedTypeSymbol>.Empty;
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
+
+        public override ImmutableArray<ParameterSymbol> Parameters =>
+            UnderlyingMethod.Parameters.SelectAsArray((Func<ParameterSymbol, NativeIntegerMethodSymbol, ParameterSymbol>)((p, m) => (ParameterSymbol)new NativeIntegerParameterSymbol(m._container, m, p)), this);
+
+        public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations =>
+            UnderlyingMethod.ExplicitInterfaceImplementations.SelectAsArray(
+                (Func<MethodSymbol, NativeIntegerTypeSymbol, MethodSymbol>)((method, map) => method.OriginalDefinition.AsMember(_container.SubstituteUnderlyingType(method.ContainingType))),
+                _container);
+
+        public override ImmutableArray<CustomModifier> RefCustomModifiers => UnderlyingMethod.RefCustomModifiers;
+
+        public override Symbol? AssociatedSymbol => _associatedSymbol;
+
+        internal override int CalculateLocalSyntaxOffset(int localPosition, SyntaxTree localTree)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override bool Equals(Symbol? other, TypeCompareKind comparison)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+            if ((object)this == other)
+            {
+                return true;
+            }
+            if (!UnderlyingMethod.Equals(other, comparison))
+            {
+                return false;
+            }
+            return (comparison & TypeCompareKind.IgnoreNativeIntegers) != 0 ||
+                other is NativeIntegerMethodSymbol;
+        }
+
+        public override int GetHashCode() => UnderlyingMethod.GetHashCode();
+
+        void Cci.IReference.Dispatch(Cci.MetadataVisitor visitor)
+        {
+            // Emit should use underlying symbol only.
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        private sealed class NativeIntegerParameterSymbol : WrappedParameterSymbol, Cci.IReference
```

Comment:
>NativeIntegerParameterSymbol [](start = 29, length = 28)

Should we override and test equality?

Reply:
No, there are multiple cases, source and PE, and there shouldn't be scenarios where we compare native integer and underlying method parameters and expect the two to be equivalent.

---
In reply to: [418345193](https://github.com/dotnet/roslyn/pull/43766#discussion_r418345193) [](ancestors = 418345193)

Reply:
> No, there are multiple cases, source and PE, and there shouldn't be scenarios where we compare native integer and underlying method parameters and expect the two to be equivalent.

Lets make sure we have a targeted test for equality of parameters.

---
In reply to: [418346367](https://github.com/dotnet/roslyn/pull/43766#discussion_r418346367) [](ancestors = 418346367,418345193)